src/components/localization/localized-link.js

```diff
@@ -155,6 +155,8 @@ const getURLFormat = (type, locale, to, affiliate_lang) => {
         return getDerivAppLocalizedURL(localized_link_url[type], locale, to)
     } else if (affiliate_links.includes(type)) {
         return `${localized_link_url[type]}?lang=${affiliate_lang}`
+    } else if (deriv_social_platforms.includes(type) && type === 'smart_trader') {
+        return `${localized_link_url[type]}/${getThaiExcludedLocale(locale)}/${to}.html`
```

Comment:
The only change that should be required for the PR should be to add `&& !deriv_other_products.includes(type)` on the conditions for `show_modal` on line 192. `smart_trader` shouldn't be categorised under social platforms.